docs/ECOS.md

```diff
@@ -0,0 +1,141 @@
+# Edge Containers
+
+## Introduction
+
+While EVE provides a lot of sophisticated device management functionality, at the end of the day its success depends on a diverse and growing ecosystem of applications that it can support. EVE is *not* a traditional operating system in that it doesn't support applications represented as executable files and [POSIX processes](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_295). Instead EVE expects its applications to be:
+
+* [Traditional Virtual Machines](https://en.wikipedia.org/wiki/Virtual_machine)
+* [Unikernels](http://unikernel.org/)
+* [Docker/OCI container](https://www.opencontainers.org)
+
+Edge Container is a novel concept that allows an effective packaging and lifecycle management of all these types of applications and it consists of two parts:
+
+* Edge Container Image (ECI) - an self-contained, binary representation of an Edge Container
+* Edge Container Object (ECO) - a live copy of an Edge Container
+
+ECOs are created by users binding a single ECI to a set of hardware and software resources (e.g. memory, CPU, I/O and networks) and we sometimes use the term ECO to refer to both -- an overall concept of an Edge Container (because it is an Edge COntainer) and a live copy. It will be fair to say that the primary purpose of EVE as an engine is to provide the best resource abstraction and an execution environment to a set of ECOs.
+
+ECI then is an application packaging and distribution mechanism that provides support for all 3 types of EVE applications while seamlessly integrating with Docker/OCI container specification and toolchains. The later part is especially important, since Docker tooling (e.g. Docker Desktop, Docker Registry, CI/CD integrations, etc.) has become a de-facto standard in modern, cloud-native application development. ECI format is a foundation for everything else in the Edge Containers workflow. This format defines the very notion of an ECO binary the same way that [ELF format](https://en.wikipedia.org/wiki/Executable_and_Linkable_Format) defines what it means to be an application on most traditional Operating Systems today. And just like with ELF, it is the job of an operating system to load (L in ELF) and execute (E in ELF) these turning them into running ECOs.
```

Comment:
Remove "then" here. Let's be definitive: "ECU is the standard application package format..."

Reply:
"distribution mechanism" - does ECI provide any distribution mechanism? Isn't ECI just the package format? Or is there more to it?

Reply:
"The later part" should be "the latter part".

Reply:
Maybe rewrite the second sentence as "Docker tooling has become a de-facto standard in modern, cloud-native application development. ECI packaging seamlessly integrates with the OCI container specification and toolchains."

Reply:
"does ECI provide any distribution mechanism?" -- yes it does -- see below